### PR TITLE
Add Font, Frame, Framesetter, Paragraph, Run, and Line test pages

### PR DESCRIPTION
**Pull request by aballway**
Add a test page to CTCatalog for CTFont, CTFrame, CTFramesetter, CTParagraph, CTRun, and CTLine.  Also has very tiny cleanup in Core Text unit tests and marks CTLineCreateWithAttributedString as stubbed properly.

### CODE REVIEW

**Comment by aballway** [comment]
Buffers

**Comment by aballway** [comment]
@rajsesh-msft is added to the review. #Closed

**Comment by aballway** [comment]
@mnithish is added to the review. #Closed

File: tests/testapps/CTCatalog/CTCatalog/CTCBaseViewController.h [hunk 1]
```diff
@@ -17,6 +17,19 @@
 #import <UIKit/UIKit.h>
 #import <CoreText/CoreText.h>
 
+// Convenience macro to create and add rows for unimplemented methods to the table
+#define ADD_UNIMPLEMENTED(ARRAY, METHOD_NAME) [ARRAY addObject:createTextCell(METHOD_NAME, @"UNIMPLEMENTED")]
+
+// Creates a table cell with title and text value
+static UITableViewCell* createTextCell(NSString* title, NSString* value) {
```
Comment (rajsesh-msft) [inline tests/testapps/CTCatalog/CTCatalog/CTCBaseViewController.h]: >static [](start = 0, length = 6)

static methods in headers don't make much sense. you could make it inline, or just have the implementation in the mm file and declaration here.

File: tests/unittests/CoreText/CTLineTests.mm [hunk 2]
```diff
@@ -27,7 +27,7 @@
     [string addAttribute:NSForegroundColorAttributeName value:[UIColor redColor] range:wholeRange];
     [string addAttribute:NSFontAttributeName value:font range:wholeRange];
 
-    return string;
+    return [string autorelease];
```
Comment (rajsesh-msft) [inline tests/unittests/CoreText/CTLineTests.mm]: >autorelease [](start = 19, length = 11)

string is already auto released...

File: tests/testapps/CTCatalog/CTCatalog/Tests/CTCFontTestViewController.mm [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+        self.tableView.allowsSelection = NO;
+        self.view.backgroundColor = [UIColor whiteColor];
+
+        // Query all available fonts and add them to array by name
+        NSMutableArray* fonts = [NSMutableArray new];
+        for (NSString* familyName in [UIFont familyNames]) {
+            for (NSString* fontName in [UIFont fontNamesForFamilyName:familyName]) {
+                [fonts addObject:fontName];
+            }
```
Comment (rajsesh-msft) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCFontTestViewController.mm]: you can replace this loop with fonts addObjectsFromArray:[UIFont fontNamesForFamilyName:familyName

File: tests/testapps/CTCatalog/CTCatalog/Tests/CTCFontTestViewController.mm [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    [_rows addObject:createTextCell(@"CTFontCopyFamilyName", (__bridge NSString*)CTFontCopyFamilyName(_font))];
+    [_rows addObject:createTextCell(@"CTFontCopyFullName", (__bridge NSString*)CTFontCopyFullName(_font))];
+    [_rows addObject:createTextCell([NSString stringWithFormat:@"CTFontCopyName:%@", _nameKey],
+                                    (__bridge NSString*)CTFontCopyName(_font, _nameKey))];
+
+    [_rows addObject:createTextCell(@"CTFontGetAscent", [NSString stringWithFormat:@"%f", CTFontGetAscent(_font)])];
+    [_rows addObject:createTextCell(@"CTFontGetDescent", [NSString stringWithFormat:@"%f", CTFontGetDescent(_font)])];
+    [_rows addObject:createTextCell(@"CTFontGetSize", [NSString stringWithFormat:@"%f", CTFontGetSize(_font)])];
+
+    // Glyph indicies for "Glyphs" in default font as CTLineCreateWithAttributdString does not currently function
```
Comment (rajsesh-msft) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCFontTestViewController.mm]: >CTLineCreateWithAttributdString [](start = 54, length = 31)

typo: CTLineCreateWithAttributedString (missing e)

File: tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+    // Make dictionary for attributed string with font, color, and alignment
+    NSDictionary* attributesDict = [NSDictionary dictionaryWithObjectsAndKeys:(__bridge id)myCFFont,
+                                                                              (id)kCTFontAttributeName,
+                                                                              color.CGColor,
+                                                                              (id)kCTForegroundColorAttributeName,
+                                                                              (__bridge id)_paragraphStyle,
+                                                                              (id)kCTParagraphStyleAttributeName,
+                                                                              nil];
+
+    NSString* text = @"The quick brown fox jumps over the lazy dog.";
```
Comment (rajsesh-msft) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm]: >dog [](start = 63, length = 3)

nit: As a dog person, I prefer "The quick brown dog jumps over the lazy fox.".

**Comment by rajsesh-msft** [comment]
:clock1:

File: tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm (hunk 5 above)
Comment (shahmeeresmail) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm]: nit: aren't foxes technically canines making them dogs as well?

File: tests/unittests/CoreText/CTLineTests.mm (hunk 2 above)
Comment (aballway) [inline tests/unittests/CoreText/CTLineTests.mm]: whoops, I think these were remnants from switching between branches a bunch.

---
In reply to: [76315512](https://github.com/Microsoft/WinObjC/pull/909#discussion_r76315512) [](ancestors = 76315512)

File: tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm (hunk 5 above)
Comment (aballway) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm]: It looks like they're both in the same family Canidae but different species.

---
In reply to: [76319688](https://github.com/Microsoft/WinObjC/pull/909#discussion_r76319688) [](ancestors = 76319688)

Comment (MSFTFox) [inline tests/testapps/CTCatalog/CTCatalog/Tests/CTCParagraphStyleTestViewController.mm]: As a Fox, I prefer the line as written.

**Comment by MSFTFox** [comment]
:shipit:

**Pull request merged**